tests/dialects/test_hive.py

```diff
@@ -459,6 +459,7 @@ def test_hive(self):
                 "hive": "MAP(a, b, c, d)",
                 "presto": "MAP(ARRAY[a, c], ARRAY[b, d])",
                 "spark": "MAP(a, b, c, d)",
+                "snowflake": "OBJECT_CONSTRUCT(a, b, c, d)",
```

Comment:
@tobymao any advice on how to be implement this? Basically I need the parser to understand that `OBJECT_CONSTRUCT` function creates a MAP value.

Reply:
What about:
```python
FUNCTIONS = {
    **parser.Parser.FUNCTIONS,
    "ARRAYAGG": exp.ArrayAgg.from_arg_list,
    "IFF": exp.If.from_arg_list,
    "TO_TIMESTAMP": _snowflake_to_timestamp,
    "ARRAY_CONSTRUCT": exp.Array.from_arg_list,
    "RLIKE": exp.RegexpLike.from_arg_list,
    "DECODE": exp.Matches.from_arg_list,
    "OBJECT_CONSTRUCT": parser.parse_var_map,
}
```

Reply:
Awesome this is perfect! Thanks!